trusted roots article split conceptual + how-to (#6893)
MicrosoftDocs/windowsserverdocs: WindowsServerDocs/identity/ad-cs/configure-trusted-roots-disallowed-certificates.md

description: Guidance on how to configure individual software updates for automatic daily Root Certificate Updates, including certificate trust lists (CTLs) and other functionality
ms.service: 
ms.topic: how-to 
author: gswashington
ms.author: roharwoo
ms.date: 01/18/2023
---

# Configure trusted roots and disallowed certificates

Applies To: Windows Server (All supported versions), Windows clients, Azure Stack HCI.

This article contains guidance intended to help administrators install automatic daily update functionality of Certificate Trust Lists (CTLs). Servers and clients access the Windows Update site to update CTLs using the automatic daily Root Certificate Updates mechanism discussed in this article.

See [Certificates and trust](certificates-and-trust.md) for more detailed knowledge on how the Microsoft Root Certificate Program works to distribute trusted root certificates automatically across Windows operating systems.

## Prerequisites

- Client machine with internet access to run certutil
- File server or web server for file storage
- AD Group policy or MDM solution to deploy configuration settings to your client

## Software update description

> The following automatic update mechanisms are available in Windows Server 2012 R2 and Windows 8.1 and newer Windows versions or when the appropriate software update is installed (for older operating systems discussed in this article):

Reviewer: ```suggestion
The following automatic update mechanisms are available in Windows Server and the Windows client when the appropriate software update is installed (for older operating systems discussed in this article):
```

- **Synchronization options** If the URL for the Windows Update site is moved to a local shared folder, the local shared folder must be synchronized with the Windows Update folder. This software update adds a set of options in the Certutil tool that administrators can use to enable synchronization. For more information, see the [New Certutil Options]() section.

- **Tool to select trusted root certificates** This software update introduces a tool for administrators who manage the set of trusted root certificates in their enterprise environment. Administrators can view and select the set of trusted root certificates, export them to a serialized certificate store, and distribute them by using Group Policy. For more information, see the [New Certutil Options]() section in this document.

- **Independent configurability** The automatic update mechanism for trusted and untrusted certificates are independently configurable. Administrators can use the automatic update mechanism to download only the untrusted CTLs and manage their own list of trusted CTLs. For more information, see the [Registry settings modified]() section in this document.

> Automatic update functionality also can be disabled if necessary, although this isn't recommended. See [Event ID 8 — Automatic Root Certificates Update Configuration](https://learn.microsoft.com/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/cc734054(v=ws.10)?redirectedfrom=MSDN#turn-off-automatic-root-certificates-update) for more information.

Reviewer: although **this** isn't recommended. - This isn't always clear to the reader when English is their second language or the text is localized. Could you use a noun or rephase?

> Automatic update functionality also can be disabled if necessary, although this isn't recommended. See [Event ID 8 — Automatic Root Certificates Update Configuration](https://learn.microsoft.com/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/cc734054(v=ws.10)?redirectedfrom=MSDN#turn-off-automatic-root-certificates-update) for more information.

Reviewer: ```suggestion
Automatic update functionality also can be disabled if necessary, although this isn't recommended. For more information, see [Event ID 8 — Automatic Root Certificates Update Configuration](https://learn.microsoft.com/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/cc734054(v=ws.10)?redirectedfrom=MSDN#turn-off-automatic-root-certificates-update) for more information.
```

- disallowedcertstl.cab, which contains a CTL with untrusted certificates

- disallowedcert.sst, which contains a serialized certificate store, including untrusted certificates

- *thumbprint*.crt, which contains non-Microsoft root certificates

The steps to perform this configuration are described in the [Configure a file or web server to download the CTL files]() section of this document.

By using Windows Server 2012 R2 and Windows 8.1 (or by installing the previously mentioned software updates on supported operating systems), an administrator can:

- Configure Active Directory Domain Services (AD DS) domain member computers to use the automatic update mechanism for trusted and untrusted CTLs, without having access to the Windows Update site. This configuration is described in the [Redirect the Microsoft Automatic Update URL for a disconnected environment]() section of this document.

- Configure AD DS domain member computers to independently opt in for untrusted and trusted CTL automatic updates. This configuration is described in the [Redirect the Microsoft Automatic Update URL for untrusted CTLs only]() section of this document.

> - Examine the set of root certificates in the Windows Root Certificate Program. This enables administrators to select a subset of certificates to distribute by using a Group Policy Object (GPO). This configuration is described in the [Use a subset of the trusted CTLs]() section of this document.

Reviewer: Starting the second sentence with this. This isn't always clear to the reader when English is their second language or the text is localized. Could you use a noun or rephase?

> - If you plan to use a web server, you should create a new virtual directory for the CTL files. The steps to create a virtual directory by using Internet Information Services (IIS) are nearly the same for all the supported operating systems discussed in this document. For more information, see <A href="https://technet.microsoft.com/library/cc771804.aspx">Create a Virtual Directory (IIS7)</A>.

- Certain system and application folders in Windows have special protection applied to them. For example, the <STRONG>inetpub</STRONG> folder requires special access permissions, which make it difficult to create a shared folder for use with a scheduled task to transfer files. As an administrator, you're typically able to create a folder location at the root of a logical drive system to use for file transfers. </LI></UL>

## Redirect the Microsoft Automatic Update URL for a disconnected environment

> If the computers in your network are configured in a domain environment and they're unable to use the automatic update mechanism or download CTLs, you can implement a GPO in AD DS to configure those computers to obtain the CTL updates from an alternate location.

Reviewer: This is quite a complex sentence. Could it be reworded to simplify?

1. In the navigation pane, under **Computer Configuration**, expand **Policies**.

1. Right-select **Administrative Templates**, and then select **Add/Remove Templates**.

1. In **Add/Remove Templates**, select **Add**. In the **Policy Templates** dialog box, select the .adm template that you previously saved. select **Open**, and then select **Close**.

1. In the navigation pane, expand **Administrative Templates**, and then expand **Classic Administrative Templates (ADM)**.

1. elect **Windows AutoUpdate Settings**, and in the details pane, double-select **URL address to be used instead of default ctldl.windowsupdate.com**.

1. Select **Enabled**. In the **Options** section, enter the URL to the file server or web server that contains the CTL files. For example, `https://server1/CTL` or `file://\\\\server1\\CTL`. select **OK**. Close the Group Policy Management Editor.

The policy is effective immediately, but the client computers must be restarted to receive the new settings, or you can type **gpupdate /force** from an elevated command prompt or from Windows PowerShell.

> [!IMPORTANT]

> > The trusted and untrusted CTLs can be updated on a daily basis, so ensure that you keep the files synchronized by using a scheduled task or another method (such as a script that handles error conditions) to update the shared folder or web virtual directory. For additional details about creating a scheduled task, see [Schedule a Task](https://technet.microsoft.com/library/cc748993.aspx). If you plan to write a script to make daily updates, see the [New Certutil Options]() and [Potential errors with Certutil -SyncWithWU]() sections of this document. These sections provide more information about command options and the error conditions.

Reviewer: ```suggestion
> The trusted and untrusted CTLs can be updated on a daily basis, so ensure that you keep the files synchronized by using a scheduled task or another method (such as a script that handles error conditions) to update the shared folder or web virtual directory. For more information about creating a scheduled task, see [Schedule a Task](https://technet.microsoft.com/library/cc748993.aspx). If you plan to write a script to make daily updates, see the [New Certutil Options]() and [Potential errors with Certutil -SyncWithWU]() sections of this document. These sections provide more information about command options and the error conditions.
```

## Redirect the Microsoft Automatic Update URL for untrusted CTLs only

> Some organizations may want only the untrusted CTLs (not the trusted CTLs) to be automatically updated. To accomplish this, you can create two .adm templates to add to Group Policy.

Reviewer: `To accomplish this` - This isn't always clear to the reader when English is their second language or the text is localized. Could you use a noun or rephase?

1. Close the Group Policy Management Editor.

The policy is effective immediately, but the client computers must be restarted to receive the new settings, or you can type **gpupdate /force** from an elevated command prompt or from Windows PowerShell.

## Registry settings modified

The settings described in this document configure the following registry keys on the client computers. These settings aren't automatically removed if the GPO is unlinked or removed from the domain. These settings must be reconfigured, if you want to change them.

| Registry keys | Value and Description |
|--------|---------|
| HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\SystemCertificates\AuthRoot\DisableRootAutoUpdate | A value of 1 disables the Windows AutoUpdate of the trusted CTL. |  
| HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\SystemCertificates\AuthRoot\EnableDisallowedCertAutoUpdate | A value of 1 enables the Windows AutoUpdate of the untrusted CTL. |
| HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\SystemCertificates\AuthRoot\AutoUpdate\RootDirUrl | Configures the shared location (the HTTP or the FILE path). |

## Deleting Trusted and Untrusted CTLs

> It may be necessary for various reasons to delete all Trusted and Untrusted CTLs from a client machine. The following Certutil options can be used to accomplish this.

Reviewer: `can be used to accomplish this.` - This isn't always clear to the reader when English is their second language or the text is localized. Could you use a noun or rephase?

- If you use a non-existent local path or folder as the destination folder, you'll see the error:
  
    The system can't find the file specified. 0x80070002 (WIN32: 2 ERROR\_FILE\_NOT\_FOUND)

- If you use a non-existent or unavailable network location as the destination folder, you'll see the error:

    The network name can't be found. 0x80070043 (WIN32: 67 ERROR\_BAD\_NET\_NAME)

- If your server can't connect over TCP port 80 to Microsoft Automatic Update servers, you'll receive the following error:

    A connection with the server couldn't be established 0x80072efd (INet: 12029 ERROR\_INTERNET\_CANNOT\_CONNECT)

- If your server is unable to reach the Microsoft Automatic Update servers with the DNS name ctldl.windowsupdate.com, you'll receive the following error:

    The server name or address couldn't be resolved 0x80072ee7 (INet: 12007 ERROR\_INTERNET\_NAME\_NOT\_RESOLVED).

- If you don't use the **-f** switch, and any of the CTL files already exist in the directory, you'll receive a *file* exists error:

    CertUtil: -syncWithWU command FAILED: 0x800700b7 (WIN32/HTTP: 183 ERROR\_ALREADY\_EXISTS) Certutil: Can't create a file when that file already exists.

> - If there's a change in the trusted root certificates, you'll see: "Warning\! Encountered the following no longer trusted roots: \<folder path\>\\\<thumbprint\>.crt. Use "-f -f" options to force the delete of the above ".crt" files. Was "authrootstl.cab" updated? If yes, consider deferring the delete until all clients have been updated."

Reviewer: `Use "-f -f" options to force the delete of the above ".crt" files.` Can we replace or provide more locational information to the word 'above'. I'm not even sure the word above needs to be there?
